2025-09-26 16:18:13,159 - __main__ - INFO - üöÄ Starting Real-Time HDFS Anomaly Detection Engine...
2025-09-26 16:18:13,162 - __main__ - INFO -  Database initialized: anomaly_detection.db
2025-09-26 16:18:13,167 - __main__ - INFO - ‚úÖ Connected to Redis at localhost:6379
2025-09-26 16:18:13,211 - httpx - INFO - HTTP Request: GET http://localhost:6333 "HTTP/1.1 200 OK"
2025-09-26 16:18:13,211 - __main__ - INFO - ‚úÖ Connected to Qdrant at localhost:6333
2025-09-26 16:18:13,211 - __main__ - INFO - üì• Loading ensemble model from ./ensemble/models/line_level_ensemble_v2/line_level_ensemble_v2_results.joblib
2025-09-26 16:18:13,886 - __main__ - INFO - ‚úÖ Loaded ensemble with 3 models
2025-09-26 16:18:13,886 - __main__ - INFO - üéØ Model weights for weighted voting:
2025-09-26 16:18:13,886 - __main__ - INFO -    dt: 0.0790
2025-09-26 16:18:13,886 - __main__ - INFO -    mlp: 0.4315
2025-09-26 16:18:13,886 - __main__ - INFO -    sgd: 0.0747
2025-09-26 16:18:13,886 - __main__ - INFO -    qdrant_similarity: 0.4148
2025-09-26 16:18:13,886 - __main__ - INFO - üì° Enhanced Kafka consumer started
2025-09-26 16:18:13,886 - __main__ - INFO - ‚úÖ Real-Time Anomaly Detection Engine ready!
2025-09-26 16:18:13,887 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=localhost:9092 <connecting> [IPv6 ('::1', 9092, 0, 0)]>: connecting to localhost:9092 [('::1', 9092, 0, 0) IPv6]
2025-09-26 16:18:13,898 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=localhost:9092 <checking_api_versions_recv> [IPv6 ('::1', 9092, 0, 0)]>: Broker version identified as 2.6
2025-09-26 16:18:13,898 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=localhost:9092 <connected> [IPv6 ('::1', 9092, 0, 0)]>: Connection complete.
2025-09-26 16:18:13,898 - kafka.consumer.subscription_state - INFO - Updating subscribed topics to: ('logs',)
2025-09-26 16:18:13,898 - __main__ - INFO - üì° Enhanced Kafka consumer listening to topic: logs
2025-09-26 16:18:13,901 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=localhost:9092 <connecting> [IPv6 ('::1', 9092, 0, 0)]>: connecting to localhost:9092 [('::1', 9092, 0, 0) IPv6]
2025-09-26 16:18:14,007 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=localhost:9092 <connected> [IPv6 ('::1', 9092, 0, 0)]>: Connection complete.
2025-09-26 16:18:14,007 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=localhost:9092 <connected> [IPv6 ('::1', 9092, 0, 0)]>: Closing connection. 
2025-09-26 16:18:14,119 - kafka.cluster - INFO - Coordinator for group/anomaly_engine_1758889093 is BrokerMetadata(nodeId='coordinator-1', host='localhost', port=9092, rack=None)
2025-09-26 16:18:14,120 - kafka.coordinator - INFO - Discovered coordinator coordinator-1 for group anomaly_engine_1758889093
2025-09-26 16:18:14,120 - kafka.coordinator.heartbeat - INFO - Starting new heartbeat thread
2025-09-26 16:18:14,121 - kafka.coordinator.consumer - INFO - Revoking previously assigned partitions set() for group anomaly_engine_1758889093
2025-09-26 16:18:14,122 - kafka.coordinator - INFO - Failed to join group anomaly_engine_1758889093: NodeNotReadyError: coordinator-1
2025-09-26 16:18:14,124 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=coordinator-1 host=localhost:9092 <connecting> [IPv6 ('::1', 9092, 0, 0)]>: connecting to localhost:9092 [('::1', 9092, 0, 0) IPv6]
2025-09-26 16:18:14,227 - kafka.coordinator - INFO - Failed to join group anomaly_engine_1758889093: NodeNotReadyError: coordinator-1
2025-09-26 16:18:14,227 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=coordinator-1 host=localhost:9092 <connected> [IPv6 ('::1', 9092, 0, 0)]>: Connection complete.
2025-09-26 16:18:14,328 - kafka.coordinator - INFO - (Re-)joining group anomaly_engine_1758889093
2025-09-26 16:18:14,335 - kafka.coordinator - INFO - Received member id kafka-python-2.2.15-51c475ad-c3a2-4e04-a7e3-5a60addac647 for group anomaly_engine_1758889093; will retry join-group
2025-09-26 16:18:14,336 - kafka.coordinator - INFO - Failed to join group anomaly_engine_1758889093: [Error 79] MemberIdRequiredError
2025-09-26 16:18:14,336 - kafka.coordinator - INFO - (Re-)joining group anomaly_engine_1758889093
2025-09-26 16:18:14,356 - kafka.coordinator - INFO - Successfully joined group anomaly_engine_1758889093 <Generation 1 (member_id: kafka-python-2.2.15-51c475ad-c3a2-4e04-a7e3-5a60addac647, protocol: range)>
2025-09-26 16:18:14,357 - kafka.coordinator - INFO - Elected group leader -- performing partition assignments using range
2025-09-26 16:18:14,380 - kafka.consumer.subscription_state - INFO - Updated partition assignment: [TopicPartition(topic='logs', partition=0)]
2025-09-26 16:18:14,380 - kafka.coordinator.consumer - INFO - Setting newly assigned partitions {TopicPartition(topic='logs', partition=0)} for group anomaly_engine_1758889093
2025-09-26 16:18:14,383 - kafka.consumer.fetcher - INFO - Resetting offset for partition TopicPartition(topic='logs', partition=0) to offset 236035.
2025-09-26 16:18:15,209 - __main__ - INFO - üóëÔ∏è Cleared 1 cached results
2025-09-26 16:18:17,241 - httpx - INFO - HTTP Request: GET http://localhost:6333/collections "HTTP/1.1 200 OK"
2025-09-26 17:56:44,881 - __main__ - INFO - üöÄ Starting Real-Time HDFS Anomaly Detection Engine...
2025-09-26 17:56:44,882 - __main__ - INFO -  Database initialized: anomaly_detection.db
2025-09-26 17:56:44,891 - __main__ - INFO - ‚úÖ Connected to Redis at localhost:6379
2025-09-26 17:56:44,948 - httpx - INFO - HTTP Request: GET http://localhost:6333 "HTTP/1.1 200 OK"
2025-09-26 17:56:44,948 - __main__ - INFO - ‚úÖ Connected to Qdrant at localhost:6333
2025-09-26 17:56:44,948 - __main__ - INFO - üì• Loading ensemble model from ./ensemble/models/line_level_ensemble_v2/line_level_ensemble_v2_results.joblib
2025-09-26 17:56:45,691 - __main__ - INFO - ‚úÖ Loaded ensemble with 3 models
2025-09-26 17:56:45,691 - __main__ - INFO - üéØ Model weights for weighted voting:
2025-09-26 17:56:45,691 - __main__ - INFO -    dt: 0.0790
2025-09-26 17:56:45,691 - __main__ - INFO -    mlp: 0.4315
2025-09-26 17:56:45,691 - __main__ - INFO -    sgd: 0.0747
2025-09-26 17:56:45,691 - __main__ - INFO -    qdrant_similarity: 0.4148
2025-09-26 17:56:45,692 - __main__ - INFO - üì° Enhanced Kafka consumer started
2025-09-26 17:56:45,692 - __main__ - INFO - ‚úÖ Real-Time Anomaly Detection Engine ready!
2025-09-26 17:56:45,693 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=localhost:9092 <connecting> [IPv6 ('::1', 9092, 0, 0)]>: connecting to localhost:9092 [('::1', 9092, 0, 0) IPv6]
2025-09-26 17:56:45,715 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=localhost:9092 <checking_api_versions_recv> [IPv6 ('::1', 9092, 0, 0)]>: Broker version identified as 2.6
2025-09-26 17:56:45,715 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=localhost:9092 <connected> [IPv6 ('::1', 9092, 0, 0)]>: Connection complete.
2025-09-26 17:56:45,716 - kafka.consumer.subscription_state - INFO - Updating subscribed topics to: ('logs',)
2025-09-26 17:56:45,716 - __main__ - INFO - üì° Enhanced Kafka consumer listening to topic: logs
2025-09-26 17:56:45,719 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=localhost:9092 <connecting> [IPv6 ('::1', 9092, 0, 0)]>: connecting to localhost:9092 [('::1', 9092, 0, 0) IPv6]
2025-09-26 17:56:45,825 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=localhost:9092 <connected> [IPv6 ('::1', 9092, 0, 0)]>: Connection complete.
2025-09-26 17:56:45,825 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=localhost:9092 <connected> [IPv6 ('::1', 9092, 0, 0)]>: Closing connection. 
2025-09-26 17:56:45,937 - kafka.cluster - INFO - Coordinator for group/anomaly_engine_1758895005 is BrokerMetadata(nodeId='coordinator-1', host='localhost', port=9092, rack=None)
2025-09-26 17:56:45,937 - kafka.coordinator - INFO - Discovered coordinator coordinator-1 for group anomaly_engine_1758895005
2025-09-26 17:56:45,937 - kafka.coordinator.heartbeat - INFO - Starting new heartbeat thread
2025-09-26 17:56:45,937 - kafka.coordinator.consumer - INFO - Revoking previously assigned partitions set() for group anomaly_engine_1758895005
2025-09-26 17:56:45,937 - kafka.coordinator - INFO - Failed to join group anomaly_engine_1758895005: NodeNotReadyError: coordinator-1
2025-09-26 17:56:45,938 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=coordinator-1 host=localhost:9092 <connecting> [IPv6 ('::1', 9092, 0, 0)]>: connecting to localhost:9092 [('::1', 9092, 0, 0) IPv6]
2025-09-26 17:56:46,042 - kafka.coordinator - INFO - Failed to join group anomaly_engine_1758895005: NodeNotReadyError: coordinator-1
2025-09-26 17:56:46,042 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=coordinator-1 host=localhost:9092 <connected> [IPv6 ('::1', 9092, 0, 0)]>: Connection complete.
2025-09-26 17:56:46,146 - kafka.coordinator - INFO - (Re-)joining group anomaly_engine_1758895005
2025-09-26 17:56:46,162 - kafka.coordinator - INFO - Received member id kafka-python-2.2.15-ed5eebdf-af54-4a8c-9cfd-fa8d407aa235 for group anomaly_engine_1758895005; will retry join-group
2025-09-26 17:56:46,162 - kafka.coordinator - INFO - Failed to join group anomaly_engine_1758895005: [Error 79] MemberIdRequiredError
2025-09-26 17:56:46,162 - kafka.coordinator - INFO - (Re-)joining group anomaly_engine_1758895005
2025-09-26 17:56:46,187 - kafka.coordinator - INFO - Successfully joined group anomaly_engine_1758895005 <Generation 1 (member_id: kafka-python-2.2.15-ed5eebdf-af54-4a8c-9cfd-fa8d407aa235, protocol: range)>
2025-09-26 17:56:46,187 - kafka.coordinator - INFO - Elected group leader -- performing partition assignments using range
2025-09-26 17:56:46,214 - kafka.consumer.subscription_state - INFO - Updated partition assignment: [TopicPartition(topic='logs', partition=0)]
2025-09-26 17:56:46,214 - kafka.coordinator.consumer - INFO - Setting newly assigned partitions {TopicPartition(topic='logs', partition=0)} for group anomaly_engine_1758895005
2025-09-26 17:56:46,220 - kafka.consumer.fetcher - INFO - Resetting offset for partition TopicPartition(topic='logs', partition=0) to offset 236035.
2025-09-26 17:56:48,851 - httpx - INFO - HTTP Request: GET http://localhost:6333/collections "HTTP/1.1 200 OK"
2025-09-26 18:05:31,098 - __main__ - INFO - üöÄ Starting Real-Time HDFS Anomaly Detection Engine...
2025-09-26 18:05:31,100 - __main__ - INFO -  Database initialized: anomaly_detection.db
2025-09-26 18:05:31,105 - __main__ - INFO - ‚úÖ Connected to Redis at localhost:6379
2025-09-26 18:05:31,154 - httpx - INFO - HTTP Request: GET http://localhost:6333 "HTTP/1.1 200 OK"
2025-09-26 18:05:31,154 - __main__ - INFO - ‚úÖ Connected to Qdrant at localhost:6333
2025-09-26 18:05:31,154 - __main__ - INFO - üì• Loading ensemble model from ./ensemble/models/line_level_ensemble_v2/line_level_ensemble_v2_results.joblib
2025-09-26 18:05:32,083 - __main__ - INFO - ‚úÖ Loaded ensemble with 3 models
2025-09-26 18:05:32,083 - __main__ - INFO - üéØ Model weights for weighted voting:
2025-09-26 18:05:32,083 - __main__ - INFO -    dt: 0.0790
2025-09-26 18:05:32,083 - __main__ - INFO -    mlp: 0.4315
2025-09-26 18:05:32,083 - __main__ - INFO -    sgd: 0.0747
2025-09-26 18:05:32,083 - __main__ - INFO -    qdrant_similarity: 0.4148
2025-09-26 18:05:32,083 - __main__ - INFO - üì° Enhanced Kafka consumer started
2025-09-26 18:05:32,083 - __main__ - INFO - ‚úÖ Real-Time Anomaly Detection Engine ready!
2025-09-26 18:05:32,084 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=localhost:9092 <connecting> [IPv6 ('::1', 9092, 0, 0)]>: connecting to localhost:9092 [('::1', 9092, 0, 0) IPv6]
2025-09-26 18:05:32,095 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=localhost:9092 <checking_api_versions_recv> [IPv6 ('::1', 9092, 0, 0)]>: Broker version identified as 2.6
2025-09-26 18:05:32,095 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=localhost:9092 <connected> [IPv6 ('::1', 9092, 0, 0)]>: Connection complete.
2025-09-26 18:05:32,096 - kafka.consumer.subscription_state - INFO - Updating subscribed topics to: ('logs',)
2025-09-26 18:05:32,096 - __main__ - INFO - üì° Enhanced Kafka consumer listening to topic: logs
2025-09-26 18:05:32,100 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=localhost:9092 <connecting> [IPv6 ('::1', 9092, 0, 0)]>: connecting to localhost:9092 [('::1', 9092, 0, 0) IPv6]
2025-09-26 18:05:32,205 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=localhost:9092 <connected> [IPv6 ('::1', 9092, 0, 0)]>: Connection complete.
2025-09-26 18:05:32,205 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=localhost:9092 <connected> [IPv6 ('::1', 9092, 0, 0)]>: Closing connection. 
2025-09-26 18:05:32,311 - kafka.cluster - INFO - Coordinator for group/anomaly_engine_1758895532 is BrokerMetadata(nodeId='coordinator-1', host='localhost', port=9092, rack=None)
2025-09-26 18:05:32,311 - kafka.coordinator - INFO - Discovered coordinator coordinator-1 for group anomaly_engine_1758895532
2025-09-26 18:05:32,311 - kafka.coordinator.heartbeat - INFO - Starting new heartbeat thread
2025-09-26 18:05:32,312 - kafka.coordinator.consumer - INFO - Revoking previously assigned partitions set() for group anomaly_engine_1758895532
2025-09-26 18:05:32,312 - kafka.coordinator - INFO - Failed to join group anomaly_engine_1758895532: NodeNotReadyError: coordinator-1
2025-09-26 18:05:32,313 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=coordinator-1 host=localhost:9092 <connecting> [IPv6 ('::1', 9092, 0, 0)]>: connecting to localhost:9092 [('::1', 9092, 0, 0) IPv6]
2025-09-26 18:05:32,419 - kafka.coordinator - INFO - Failed to join group anomaly_engine_1758895532: NodeNotReadyError: coordinator-1
2025-09-26 18:05:32,419 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=coordinator-1 host=localhost:9092 <connected> [IPv6 ('::1', 9092, 0, 0)]>: Connection complete.
2025-09-26 18:05:32,522 - kafka.coordinator - INFO - (Re-)joining group anomaly_engine_1758895532
2025-09-26 18:05:32,538 - kafka.coordinator - INFO - Received member id kafka-python-2.2.15-dc9ac12b-a14b-4043-a3e8-d69f1036fd57 for group anomaly_engine_1758895532; will retry join-group
2025-09-26 18:05:32,538 - kafka.coordinator - INFO - Failed to join group anomaly_engine_1758895532: [Error 79] MemberIdRequiredError
2025-09-26 18:05:32,538 - kafka.coordinator - INFO - (Re-)joining group anomaly_engine_1758895532
2025-09-26 18:05:32,556 - kafka.coordinator - INFO - Successfully joined group anomaly_engine_1758895532 <Generation 1 (member_id: kafka-python-2.2.15-dc9ac12b-a14b-4043-a3e8-d69f1036fd57, protocol: range)>
2025-09-26 18:05:32,556 - kafka.coordinator - INFO - Elected group leader -- performing partition assignments using range
2025-09-26 18:05:32,570 - kafka.consumer.subscription_state - INFO - Updated partition assignment: [TopicPartition(topic='logs', partition=0)]
2025-09-26 18:05:32,570 - kafka.coordinator.consumer - INFO - Setting newly assigned partitions {TopicPartition(topic='logs', partition=0)} for group anomaly_engine_1758895532
2025-09-26 18:05:32,574 - kafka.consumer.fetcher - INFO - Resetting offset for partition TopicPartition(topic='logs', partition=0) to offset 236035.
2025-09-26 18:05:35,450 - httpx - INFO - HTTP Request: GET http://localhost:6333/collections "HTTP/1.1 200 OK"
